### ISSUE TITLE
Always-on dropout layer

### ISSUE THREAD

**Issue by janosh** [tensorflow/tensorflow#28484]
**Describe the feature and the current behavior/state.**

Add the ability to create always-on dropout layers.

**Will this change the current api? How?**

Yes, adds a boolean option `always_on` to `tf.keras.layers.Dropout`.

Currently, implementing this functionality requires writing a custom dropout layer:

```py
class Dropout(tf.keras.layers.Layer):
    """Always-on dropout layer which doesn't respect the `training` flag
    set to true by model.fit and false by model.predict.
    Unlike tf.keras.layers.Dropout, this layer does not return input
    unchanged if `training=true`, but always randomly drops a fraction 
    of the input nodes specified by `self.size`.
    """

    def __init__(self, rate, **kwargs):
        super().__init__(**kwargs)
        self.rate = rate

    def call(self, inputs):
        return tf.nn.dropout(inputs, self.rate)
```

**Who will benefit from this feature?**

Everyone using dropout to create cheap Bayesian neural network as per [this paper](https://arxiv.org/abs/1703.04977). In particular, users of `tfp` may be interested.

**Comment by tanzhenyu** [comment]
can this be part of tfp?

**Comment by janosh** [comment]
Certainly. Should I open a new issue there and mention this one?

**Comment by tanzhenyu** [comment]
Yep, that'd be great! https://github.com/tensorflow/probability

**Comment by janosh** [comment]
Done.
